### ISSUE TITLE
Spec platform information is inferred when parsing

### ISSUE THREAD

**Issue by alalazo** [LLNL/spack#22800]
If a spec string with partial architecture (platform + OS + target) information is parsed, information on its platform is added based on the current host if not present. I would expect the platform to remain `None` if further information is not given. 

### Steps to reproduce the issue

```python
>>> import spack.spec
>>> s = spack.spec.Spec('zlib target=x86_64')
>>> str(s)
'zlib arch=linux-None-x86_64'
>>> s.architecture.platform
'linux'
```

### Error Message

There's no error message, but I would expect:
```python
>>> import spack.spec
>>> s = spack.spec.Spec('zlib target=x86_64')
>>> str(s)
'zlib arch=None-None-x86_64'
>>> s.architecture.platform
'None'
```

### Information on your system

This happens regardless of the system. The line that does this is:

https://github.com/spack/spack/blob/1ed44f20f255f63b9e75e739a8bbe6911f4e1664/lib/spack/spack/spec.py#L4657

by calling this function:

https://github.com/spack/spack/blob/1ed44f20f255f63b9e75e739a8bbe6911f4e1664/lib/spack/spack/spec.py#L1249-L1259

### Additional information

This issue is relevant in the context of bootstrapping from binaries, where we may want to check if something installed or available matches a spec string that may not be built on the current architecture (but is compatible, e.g. `linux` and `cray`).

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have searched the issues of this repo and believe this is not a duplicate
- [x] I have run the failing commands in debug mode and reported the output

**Comment by alalazo** [comment]
Commenting the call to this function seems to be almost fine with the new concretizer, but causes several failures with the old concretizer.

**Comment by alalazo** [comment]
Fixed at some point (65556eb53e184e78aaacce2a9e325ea4507bca2c):
```python
Spack version 0.20.0.dev0
Python 3.8.10, Linux x86_64
>>> import spack.spec
>>> s = spack.spec.Spec('zlib target=x86_64')
>>> str(s)
'zlib arch=None-None-x86_64'
```
